File: java/src/main/java/util/AuthorNameNormalizer.java
package util;

public class AuthorNameNormalizer {
    public String normalize(String name) {
        String[] parts = parts(name);
        if (isMononym(parts))
            return name;
        return lastName(parts) + ", " +
               firstName(parts);
    }

    private String[] parts(String name) {
        return name.trim().split(" ");
    }

    private String firstName(String[] parts) {
        return parts[0];
    }

    private String lastName(String[] parts) {
        return parts[1];
    }

    private boolean isMononym(String[] parts) {
        return parts.length == 1;
    }
}

File: objc/NameNormalizerObjC/NameNormalizer/AuthorNameNormalizer.m
#import "AuthorNameNormalizer.h"

@implementation AuthorNameNormalizer

- (NSString *)normalize:(NSString *)name
{
    NSString *trimmed = [name stringByTrimmingCharactersInSet:NSCharacterSet.whitespaceCharacterSet];
    NSArray<NSString *> *parts = [trimmed componentsSeparatedByString:@" "];
    if (parts.count < 2)
        return name;
    return [NSString stringWithFormat:@"%@, %@%@",
                                      [self lastNameFromParts:parts],
                                      [self firstNameFromParts:parts],
                                      [self middleInitialFromParts:parts]];
}

- (NSString *)firstNameFromParts:(NSArray<NSString *> *)parts
{
    return parts.firstObject;
}

- (NSString *)lastNameFromParts:(NSArray<NSString *> *)parts
{
    return parts.lastObject;
}

- (NSString *)middleInitialFromParts:(NSArray<NSString *> *)parts
{
    if (parts.count > 2)
        return [NSString stringWithFormat:@" %@.", [parts[1] substringToIndex:1]];
    return @"";
}

@end
